### ISSUE TITLE
`u + a` selector is minified to `u+a`

### ISSUE THREAD

**Issue by romainmenke** [parcel-bundler/lightningcss#289]
```css
u + a {
  color: green;
}
```

becomes :

```css
u+a{color:green}
```

I expected `u + a` to avoid forming `<urange>` tokens.

https://www.w3.org/TR/2021/CRD-css-syntax-3-20211224/#urange-syntax

There are likely more error cases.

**Comment by devongovett** [comment]
hmm interesting case. I think this is correct though. It looks like a bug in Chrome. In both Firefox and Safari, the link is green.

`<urange>` is not a token type. It is composed of other tokens (ident, number, etc.). Therefore, unless explicitly called for in a property or value grammar, `<urange>` should not be parsed.

**Comment by romainmenke** [comment]
ha, I was just thinking that I should check more than just Chrome.

Could it be that Safari and Firefox are lagging behind?
I think this might have been added to the spec in the 2019 version : https://www.w3.org/TR/2021/CRD-css-syntax-3-20211224/#changes-CR-20140220

**Comment by romainmenke** [comment]
https://wpt.fyi/results/css/css-syntax/urange-parsing.html?label=master&label=experimental&aligned&view=subtest&q=urange

https://github.com/web-platform-tests/wpt/blob/09b8d2f86a/css/css-syntax/urange-parsing.html

If I read those tests correctly the parser/tokenizer should only form `<urange>` in certain contexts.

- Chrome is incorrectly parsing `u+a {}`.
- Safari and Firefox are correct.

**Comment by devongovett** [comment]
If so seems like the other way? That was removing a token. Chrome seems to still be treating it as one. Or maybe it's not related to urange at all, just another bug. 

I'd file a bug report and see what they say. You could also look in the web platform tests, or ask the CSS workiNg group.

**Comment by romainmenke** [comment]
Given the WPT tests I think it might make sense for `lightningcss` to avoid minifying as `u+a` but only to work around the Chrome bug. Not because it is actually invalid.

**Comment by romainmenke** [comment]
There is also already a chromium bug related to those WPT tests : https://bugs.chromium.org/p/chromium/issues/detail?id=1215071&q=urange%20css&can=2

Leaving it up to you to ignore this issue or maybe apply a patch.
I don't have any real world bug that requires fixing.

**Comment by devongovett** [comment]
Thanks for digging those up. I'll look into doing a workaround.

**Comment by alexander-akait** [comment]
I agree with @devongovett, sounds like a bug in chrome becaue https://github.com/parcel-bundler/lightningcss/issues/289#issuecomment-1248286049, also the same issue was in firefox bug tracker and it was marked as a bug and fixed

**Comment by yisibl** [comment]
I tested this and found it was fixed in [Chrome 118](https://chromiumdash.appspot.com/commit/05f8867a086e119885a11b95bc10614c8e7581ff).
https://issues.chromium.org/issues/40769707#comment5

```html
data:text/html;charset=UTF-8,<!doctype html>
<style>
  .u + a{color:red}
  .u+a{color:green}

  .u + b{color:red}
  .u+b{color:green}
</style>
<u class="u">Underlined</u>
<a href="">A link</a>

<u class="u">Underlined</u>
<b>bbb</b>
```
<img width="844" height="362" alt="Image" src="https://github.com/user-attachments/assets/51cd4d24-e8ef-4094-a510-76cff5a478f3" />

**Comment by yisibl** [comment]
@devongovett @alexander-akait 
What do you think about changing `u+` compression to `u +` or `u+ ` when encountered in selectors?

```css
.u+b{color:green}
:is(u+b){color:green}
/* to */
.u+ b{color:green}
:is(u+ b){color:green}
```
